### ISSUE TITLE
How to continue training pre-trained models of music_vae?

### ISSUE THREAD

**Issue by xiaoyisha** [tensorflow/magenta#1253]
I want to continue training pre-trained MusicVAE models, but I failed to find a way to do so. I know that I can load models using TrainedModel, but there is no such function that can let me continue training. I also tried train a model from scratch for a while and then substitute the .ckpt files with the pre-trained one. But doing so would cause a VariableNotFound Error during restoring.

**Comment by adarob** [comment]
Which checkpoint are you trying to use? I may have stripped out some of the training variables but I can fix that if needed.

**Comment by xiaoyisha** [comment]
The trios_16bar_hierdec.ckpt downloaded from colab.

**Comment by xiaoyisha** [comment]
by the way, what is the loss of the last step of this model?

**Comment by xiaoyisha** [comment]
Yes.

**Comment by xiaoyisha** [comment]
Yes.

**Comment by xiaoyisha** [comment]
Yes.

**Comment by xiaoyisha** [comment]
Yes.

**Comment by xiaoyisha** [comment]
Yes.

**Comment by xiaoyisha** [comment]
Yes.

**Comment by xiaoyisha** [comment]
Yes.

**Comment by xiaoyisha** [comment]
Yes.

**Comment by xiaoyisha** [comment]
Yes.

**Comment by xiaoyisha** [comment]
Yes.

**Comment by xiaoyisha** [comment]
Yes.

**Comment by xiaoyisha** [comment]
Yes.

**Comment by xiaoyisha** [comment]
Yes.

**Comment by xiaoyisha** [comment]
Yes.

**Comment by adarob** [comment]
Thanks, I will fix this tomorrow.

**Comment by xiaoyisha** [comment]
Thank you very much.

**Comment by adarob** [comment]
Can you try:
https://storage.googleapis.com/download.magenta.tensorflow.org/temp/trio_16bar_hierdec.ckpt.data-00000-of-00001
https://storage.googleapis.com/download.magenta.tensorflow.org/temp/trio_16bar_hierdec.ckpt.index

The training loss should be ~270.

**Comment by howardlau1999** [comment]
it works! thank you very much!

**Comment by jincong0623** [comment]
Hi, @adarob 
I was training the musicvae model, and batch size=32, but it's too slow just generate 50 model.ckpt files for 3 days. However, I'm not sure how to open the ckpt files or switch them to midi files and normally most of the generated files are midi files. Could you offer me some references or instruct me how to resolve the ckpt files to midi files. Thanks a lot!

**Comment by adarob** [comment]
I've updated the checkpoints.tar.gz file, which resolves this issue. @jincong0623 please open a separate issue. As a quick answer the contain model weights, not MIDI files. I have not added a script to generate yet (see #1199) but you can see how to do it in https://g.co/magenta/musicvae-colab.
